django/db/migrations/loader.py

```diff
@@ -316,6 +319,33 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
 
+    def check_consistent_history(self, connection):
+        """
+        Checks that all migrations are applied after their dependencies. Raises
+        InconsistentMigrationHistory exception when migration in applied before
+        it's dependency
+        """
+        recorder = MigrationRecorder(connection)
+        applied = recorder.applied_migrations()
+        for migration in applied:
+            # If we don't know anything about this migration we skip it
+            if migration not in self.graph.nodes:
+                continue
+            unapplied_parents = [
+                x.key for x in self.graph.node_map[migration].parents
+                if x not in applied and settings.MIGRATION_MODULES.get(x[0])
```

Comment:
I don't think this is correct. `settings.MIGRATION_MODULES` only contains user-defined migration modules -- presumably want the detection to work regardless of whether or not that setting is defined.